### ISSUE TITLE
False-positive `Non-exhaustive match` on empty enum

### ISSUE THREAD

**Issue by ortem** [intellij-rust/intellij-rust#6211]
## Environment

* **IntelliJ Rust plugin version:** 0.3.133.3389-203-nightly
* **Rust toolchain version:** 1.49.0-nightly (beb5ae474 2020-10-04) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.3 EAP Ultimate Edition (IU-203.4203.26)
* **Operating system:** macOS 10.15.6

## Problem description
The following code copiles fine but highlighted as error
<img width="463" alt="Screenshot 2020-10-07 at 08 45 58" src="https://user-images.githubusercontent.com/4854600/95292160-a22f3580-0879-11eb-84a3-ce97c9dc4bfe.png">

## Steps to reproduce

```Rust
enum E {}

fn foo(e: E) {
    match e {
    }
}
```
